List users

blueJayWays opened the pull request:
## Summary

- Add endpoint to get a list of users

- Add test cases for endpoint to get all users

## Trello Card

https://trello.com/c/60GaXrsX/4-lista-de-usuarios.

blueJayWays committed:
Merge sign-in into list-users

blueJayWays committed:
Add middleware to check if user is authenticated

blueJayWays committed:
Add endpoint to get a list of users

blueJayWays committed:
Add test cases for endpoint to get all users

blueJayWays committed:
Merge master into list-users

igna92ts commented on app/controllers/user_controller.js:
```diff
@@ -19,14 +19,41 @@ exports.signUp = (req, res, next) => {
+
+exports.getAllUsers = (req, res, next) => {
+  const limit = 5;
+  let offset = 0;
```
why a let here? cant you just create it below as a const?

igna92ts commented on app/middlewares/validations.js:
```diff
@@ -1,6 +1,21 @@
+exports.isAuthenticated = (req, res, next) => {
+  const token = req.headers['x-access-token'];
+  if (token) {
+    jwt.verify(token, config.common.session.secret, function(err, decoded) {
```
use arrow functions

igna92ts commented on app/models/user.js:
```diff
@@ -23,10 +23,17 @@ module.exports = (sequelize, DataTypes) => {
     return bcrypt.compareSync(password, this.password);
   };
 
+  User.prototype.toJSON = function() {
```
dont overwrite object prototypes

blueJayWays replied:
which is the best way to hide the encrypted password that is sent when the user is returned?

igna92ts commented on test/user.js:
```diff
@@ -4,7 +4,30 @@ const chai = require('chai'),
-  user = { name: 'name', lastname: 'lastname', email: 'test@wolox.co', password: '12345678' };
+  user = { name: 'name', lastname: 'lastname', email: 'test@wolox.co', password: '12345678' },
+  cookie = require('cookie'),
+  createUser = _ =>
```
why _ ? if it doesnt take any arguments just do ()

blueJayWays committed:
Add endpoint to get a list of users

Add test cases for endpoint to get all users

Put validationResultHandler in middlewares

Merge branch 'sign-up' into test-documentation

Add test to count the users in the DB

Update docs

Merge pull request #2 from wolox-training/sign-up

Sign up
Merge pull request #4 from wolox-training/test-documentation

Test documentation
Merge pull request #5 from wolox-training/sign-in

Sign in
Merge master into list-users

blueJayWays committed:
Apply corrections indicated in PR #6

igna92ts commented on app/controllers/user_controller.js:
```diff
@@ -19,14 +19,40 @@ exports.signUp = (req, res, next) => {
+exports.getAllUsers = (req, res, next) => {
+  const limit = 5;
+  const page = req.query.page || 1;
+  User.findAndCountAll()
```
if you are going to do this, just filter them manually  so you dont look them up twice in the db

igna92ts commented on test/user.js:
```diff
@@ -4,7 +4,30 @@ const chai = require('chai'),
+  chai
+    .request(app)
+    .post('/users/sessions')
+    .send({ email: user.email, password: user.password });
```
no need to do this, just do `.send(user)`

igna92ts commented on test/user.js:
```diff
@@ -4,7 +4,30 @@ const chai = require('chai'),
+  const cookies = res.headers['set-cookie'];
+  let token;
+  if (cookies) {
+    cookies.forEach(function(icookie) {
```
why a forEach?

blueJayWays replied:
because maybe the header can have more than one cookie, so I iterate over the cookies and get the x-access-token cookie. And the library 'cookie' helps me to parse the cookie and get just the value that I want

igna92ts commented on test/user.js:
```diff
@@ -258,3 +276,122 @@ describe('/users/sessions POST', () => {
+          .end(function(err, response) {
+            expect(response).to.have.status(200);
+            response.should.be.json;
+            response.body.should.be.a('object');
```
no need for this check, just check thats its truthy

blueJayWays committed:
Modify pagination in get all users endpoint

blueJayWays committed:
Remove test to check of response is a object
